### PR TITLE
[v3] Upgrade Browsershot

### PR DESCRIPTION
**Pull request by juukie**
Hi @freekmurze, this is what I had in mind. It allows users to set a custom created Browsershot instance via `setBrowsershot()`. What do you think?

### CODE REVIEW

**Comment by juukie** [comment]
I'm not sure how to let Travis install puppeteer. Something like
```
before_script:
  - npm install -g puppeteer
  - ./tests/server/start_server.sh
```
?

File: src/Crawler.php [hunk 1]
```diff
@@ -347,15 +345,29 @@ protected function addtoDepthTree(Node $node, string $url, string $parentUrl)
...
         }
 
-        $html = $browsershot->bodyHtml();
+        $this->browsershot = new Browsershot();
 
-        return html_entity_decode($html);
+        return $this->browsershot;
+    }
+
+    protected function setBrowsershot(Browsershot $browsershot)
```
Comment (freekmurze) [inline src/Crawler.php]: Shouldn't this method be `public`?

**Comment by freekmurze** [comment]
> I'm not sure how to let Travis install puppeteer and use Node 7 since that's what is required by Browsershot.

The tests for this PR seem to be running fine, no?

**Comment by juukie** [comment]
Thanks for looking it over. I might also add a test to cover the setBrowsershot() method.
Visibility has changed to public.

Travis issue was solved later on so that's ok :).

**Comment by juukie** [comment]
I've added the test. Although it passes, it does not really test if our custom instance is being used. 
Is there a way to check if `setBrowsershot()` really changes the $browsershot property? Since it's a protected property, I can't use `assertSame($browsershot, $crawler->browsershot)`. Any idea on this?

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**